### PR TITLE
Additional visual fixes for anywhere flyout

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description

1. Fixed so vislayers aren't mandatory for assocation to be created.
2. Various UI fixes such as keeping link text from wrapping, not letting association detector button expand, changing link to open in new tab and aligning some of the text in flyout. 

- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx [hunk 1]
```diff
@@ -305,16 +305,18 @@ function AssociatedDetectors({ embeddable, closeFlyout, setMode }) {
...
-                fill
-                iconType="link"
-                onClick={() => {
-                  setMode('existing');
-                }}
-              >
-                Associate a detector
-              </EuiButton>
+            <EuiFlexItem grow={false}>
+              <div>
```
Comment (kaituo) [inline public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx:309]: Why do you need div inside the EuiFlexItem?

**Comment by kaituo** [comment]
unit tests failed as well:

Summary of all failing tests
FAIL public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/__tests__/AssociatedDetectors.test.tsx
  ● Test suite failed to run

    Cannot find module '../../../../../../../../src/plugins/vis_augmenter/public/utils' from 'public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/__tests__/AssociatedDetectors.test.tsx'

      91 | }));
      92 |
    > 93 | jest.mock(
         |      ^
      94 |   '../../../../../../../../src/plugins/vis_augmenter/public/utils',
      95 |   () => ({
      96 |     getAugmentVisSavedObjs: jest.fn(),

      at Resolver.resolveModule (../../node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/__tests__/AssociatedDetectors.test.tsx:93:6)

FAIL public/expressions/__tests__/overlay_anomalies.test.ts
  ● Test suite failed to run

    Cannot find module '../../../../src/plugins/vis_augmenter/public' from 'public/expressions/helpers.ts'

    Require stack:
      public/expressions/helpers.ts
      public/expressions/__tests__/overlay_anomalies.test.ts

      11 | import { AnomalyData } from '../models/interfaces';
      12 | import { getClient } from '../services';
    > 13 | import {
         | ^
      14 |   PluginResource,
      15 |   PointInTimeEventsVisLayer,
      16 |   VisLayerError,

      at Resolver.resolveModule (../../node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (public/expressions/helpers.ts:13:1)


Test Suites: 2 failed, 1 skipped, 74 passed, 76 of 77 total
Tests:       2 skipped, 349 passed, 351 total
Snapshots:   70 passed, 70 total
Time:        310.59 s

**Comment by amitgalitz** [comment]
> unit tests failed as well:

unit tests pass locally on my end, they are failing here because we are testing against main branch in OSD but changes are still in a feature branch in OSD. Initially didn't want to change the github workflow cause I thought it would be merged to main earlier and that we would also be merging our code to main from featureAnywhere soon. I can change the workflow and see if possible to test against feature branch though if you prefer that.

File: public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx (hunk 1 above)
Comment (kaituo) [reply public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx:309]: got it

**Review by kaituo — APPROVED** [review]

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**